Helper/Order.php

```diff
@@ -355,4 +356,20 @@ public function has_childs($parent)
     {
         return isset($parent['operation_type']) && $parent['operation_type'] == 'payment.multiple-sources';
     }
+
+    /**
+     * Maybe add product subscriptions sign-up fee 
+     * 
+     * @param object $checkout used to get items and total
+     * 
+     * @return int|string total cleared
+     */
+    public function maybe_add_signup_fee($checkout)
+    { 
+        foreach ($checkout->items as $item)
+            if($item['type'] == 'subscription') 
+                $checkout->set_signup_fee($this->order->get_total_fees());
```

Comment:
Siento que esto da una vuelta de más. No sería más conveniente actualizar el total con directamente con `$checkout->total -= $this->order->get_total_fees();`. Por otro lado, cuidado con usar `$this->order->get_total_fees()` porque también te va a traer las fees que no tengan nada que ver con suscripciones. Ej. si agregan una comisión por país.

Reply:
Bien, entonces puedo hacer algo intermedio.
Eliminar set_signup_fee, pero seguir filtrando por tipo de item para no añadir otros fees. Asi se elimina buena parte del código relacionado a guardar valores en la prop y directamente restamos los fees que necesitamos direcamente del total como recomendaste